### ISSUE TITLE
[BUG] Hover tool offset when rendered in Jupyter-notebook and sizing issue

### ISSUE THREAD

**Issue by weinbe58** [bokeh/bokeh#13487]
### Software versions

Python version      :  3.10.13 (main, Aug 24 2023, 22:36:46) [Clang 14.0.3 (clang-1403.0.22.14.1)]
IPython version     :  8.16.1
Tornado version     :  6.3.3
Bokeh version       :  3.3.0
BokehJS static path :  /Users/pweinberg/Documents/Bloqade_dev/Symbolics/bloqade-python-2/.venv/lib/python3.10/site-packages/bokeh/server/static
node.js version     :  v20.5.1
npm version         :  9.8.0
Operating system    :  macOS-13.5.1-arm64-arm-64bit

### Browser name and version

Google Chrome

### Jupyter notebook / Jupyter Lab version

4.0.5

### Expected behavior

The plot should render and the hover tool should align with the position of the points on the graph in the interactive plot and display the current (x,y) coordinate on the curve.

### Observed behavior

The hover tool appears to be offset from where it should be. 

Also when rendering the plot with dynamic sizing sometimes it will get stuck in a loop where it will try to scale to the width of the output cell but then a scroll bar will appear and the plot will scale down again and this will repeat. This weird scaling behavior only happens for certain cell sizes and it also depends on the aspect ratio. 

### Example code

```Python
In a Jupyter-notebook run the following code in a single cell. 


from bokeh.io import output_notebook
from bokeh.plotting import figure, show
from bokeh.layouts import column
from bokeh.models import ColumnDataSource, HoverTool
import random 

output_notebook()


times = sorted([random.uniform(0,1) for i in range(10)])
values = sorted([random.uniform(0,1) for i in range(10)])

data_source = ColumnDataSource(data=dict(times=times, values=values))

fig = figure(x_axis_label="times", y_axis_label="values", sizing_mode="scale_both", aspect_ratio=2)

fig.line(x="times", y="values", source=data_source)
fig.scatter(x="times", y="values", source=data_source)

hov_tool = HoverTool(mode="vline")
hov_tool.tooltips = [("(x,y)", "(@times, @values)")]
fig.add_tools(hov_tool)

show(fig)
```

### Screenshots

<img width="1155" alt="Screenshot 2023-10-25 at 10 14 06 AM" src="https://github.com/bokeh/bokeh/assets/10360423/d5741075-ed26-4e66-8f93-07a8a2fbfcdb">

**Comment by kuatroka** [comment]
After updating from 3.2.2 to 3.3.0 and using bokeh as the backend in the `hvplot` library, I don't get the hover at all. With the 3.2.2 it works fine.

I think these cases are related.

**Comment by mattpap** [comment]
This was fixed in PR #13431 and will be released next week as a part of bokeh 3.3.1.

**Comment by kuatroka** [comment]
Great
